src/lib/y2storage/proposal/autoinst_devices_planner.rb

```diff
@@ -246,17 +245,26 @@ def add_snapshots(device, drive_section)
       def add_subvolumes_attrs(device, section)
         return unless device.btrfs?
 
-        subvol_specs = section.subvolumes
-        mount = device.mount_point
+        defaults = subvolume_attrs_for(device.mount_point)
 
-        if subvol_specs.nil? && mount == "/"
-          @default_subvolumes_used = true
-          subvol_specs = proposal_settings.subvolumes
-        end
+        device.default_subvolume = section.subvolumes_prefix || defaults[:subvolumes_prefix]
 
-        device.default_subvolume = section.subvolumes_prefix ||
-          proposal_settings.legacy_btrfs_default_subvolume
-        device.subvolumes = subvol_specs
+        device.subvolumes =
+          if section.create_subvolumes
+            section.subvolumes || defaults[:subvolumes] || []
```

Comment:
NP: Why to not always ensure useful defaults? I mean, here I would expect that `defaults[:subvolumes]` always has a proper default value, but sometimes it is `nil`, which isn't a valid default for subvolumes (it needs to be an array).

Reply:
Because if you use a `Btrfs` filesystem on a partition which is not defined in the list of volumes, it won't work.